lib/internal/repl.js

```diff
@@ -51,7 +53,7 @@ function createRepl(env, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY_FILE) {
+  if (opts.terminal) {
```

Comment:
The "we can't write to this file at all" logic should handle the invalid path case.